group/assignment5/assignment5_graph.py

```diff
@@ -0,0 +1,38 @@
+from collections import defaultdict
+from os.path import commonprefix
+ 
+class Graph:
+    def __init__(self, dictionary):
+        self.graph = defaultdict(list)
+        self.letters = set()
+        for first, second in zip(dictionary, dictionary[1:]):
+            prefix = commonprefix([first, second])
```

Comment:
I would recommend to not mix processing of the input and constructing the graph. That would make your graph more flexible and reusable.